attest/attest.go

```diff
@@ -234,10 +234,14 @@ func ParseAKPublic(version TPMVersion, public []byte) (*AKPublic, error) {
 		if err != nil {
 			return nil, fmt.Errorf("parsing TPM public key structure: %v", err)
 		}
+		if pub.RSAParameters.Sign == nil {
```

Comment:
One question, what if this is a EC key?

Reply:
Would the check be better in the switch statement below?

Reply:
Added validation for both RSA & ECC parameters.